src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -101,15 +99,15 @@ class DoorLockClusterFabricDelegate : public chip::FabricTable::Delegate
         }
     }
 };
-static DoorLockClusterFabricDelegate gFabricDelegate;
 
 /**********************************************************
  * DoorLockServer public methods
  *********************************************************/
 
 DoorLockServer & DoorLockServer::Instance()
 {
-    return instance;
+    static DoorLockServer sInstance;
```

Comment:
Why is this better than a global? It introduces a slight overhead due to lazy initialization of static local variables and C++-imposed thread safety of the initialization, which may be justified but what's really the goal of this change?